tekton-catalog/tekton-kfptask/pkg/reconciler/kfptask/reconciler.go

```diff
@@ -63,13 +73,34 @@ type Reconciler struct {
 // borrow from kubeflow/pipeline/backend/src/common/util
 const LabelKeyWorkflowRunId = "pipeline/runid"
 
-// for kfptaskState
+// for kfptaskState and driver
 const (
 	StateInit = iota
 	StateRunning
 	StateDone
+	// ReasonFailedValidation indicates that the reason for failure status is that Run failed runtime validation
+	ReasonFailedValidation = "RunValidationFailed"
+
+	// ReasonDriverError indicates that an error is throw while running the driver
+	ReasonDriverError = "DriverError"
+
+	// ReasonDriverSuccess indicates that driver finished successfully
+	ReasonDriverSuccess = "DriverSuccess"
+	ReasonTaskCached    = "TaskCached"
+
+	ExecutionID    = "execution_id"
+	ExecutorInput  = "executor_input"
+	Condition      = "condition"
+	IterationCount = "iteration-count"
```

Comment:
I am thinking could this be also changed to `iteration_count`. This is used as the ResultSet's name. Also need to change compiler as well.

Reply:
maybe not. I just checked the compiler part. there are other Iteration related params also need to change if you update this one. So maybe leave this as-is